src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClient.cs

```diff
@@ -266,20 +230,21 @@ public List<PSRoleAssignment> FilterRoleAssignments(FilterRoleAssignmentsOptions
                         throw new InvalidOperationException(ProjectResources.ExpandGroupsNotSupported);
                     }
 
-                    parameters.AssignedToPrincipalId = adObject.Id;
-                }
+					parameters = new ODataQuery<RoleAssignmentFilter>(f => f.PrincipalId == adObject.Id.ToString());
+				}
                 else
                 {
-                    parameters.PrincipalId = string.IsNullOrEmpty(options.ADObjectFilter.Id) ? adObject.Id : Guid.Parse(options.ADObjectFilter.Id);
-                }
+					string principalId = string.IsNullOrEmpty(options.ADObjectFilter.Id) ? adObject.Id.ToString() : options.ADObjectFilter.Id;
+					parameters = new ODataQuery<RoleAssignmentFilter>(f => f.PrincipalId == principalId);
```

Comment:
many of these changes are done as part of the code which is already present in the repo,do a rebase of the code,so that merge conflicts wont arise later